types.go

```diff
@@ -30,6 +30,9 @@ type SparseMerkleTree interface {
 	Root() []byte
 	// Prove computes a Merkle proof of membership or non-membership of a key.
 	Prove(key []byte) (*SparseMerkleProof, error)
+	// ProveClosest computes a Merkle proof of inclusion for a key in the tree which is
+	// closest to the path provided
```

Comment:
Add details on what closest means.

E.g. Most overlap in the path (i.e. # of bits) + default to left?